pluggy.py

```diff
@@ -679,6 +702,13 @@ def __repr__(self):
 
     def __call__(self, **kwargs):
         assert not self.is_historic()
+        notincall = set(self.argnames) - set(kwargs.keys())
+        if notincall:
+            warnings.warn(
+                "Positional arg(s) %s are declared in the hookspec "
```

Comment:
This one I agree with since we only allow calling with "named argument" syntax currently.